src/formatDate.js

```diff
@@ -50,7 +50,37 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(`${fromFormat[3]}`);
+  const dateObj = {};
+  let finalDate = '';
+
+  for (let i = 0; i <= dateArr.length; i++) {
+    dateObj[fromFormat[i]] = dateArr[i];
+  }
+
+  if (toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(2);
+  }
+
+  if (toFormat.includes('YYYY')) {
+    if (!dateObj.hasOwnProperty('YYYY')) {
```

Comment:
And will be nice to create a separate constant for it

```suggestion
  if (!dateObj.hasOwnProperty('YYYY') && toFormat.includes('YYYY')) {
```